presto-tests/src/test/java/com/facebook/presto/execution/TestBeginQuery.java

```diff
@@ -0,0 +1,315 @@
+import com.facebook.presto.spi.ConnectorHandleResolver;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.FixedPageSource;
+import com.facebook.presto.spi.Plugin;
+import com.facebook.presto.spi.connector.Connector;
+import com.facebook.presto.spi.connector.ConnectorContext;
+import com.facebook.presto.spi.connector.ConnectorFactory;
+import com.facebook.presto.spi.connector.ConnectorMetadata;
+import com.facebook.presto.spi.connector.ConnectorPageSinkProvider;
+import com.facebook.presto.spi.connector.ConnectorPageSourceProvider;
+import com.facebook.presto.spi.connector.ConnectorSplitManager;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.transaction.IsolationLevel;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.testing.TestingHandleResolver;
+import com.facebook.presto.testing.TestingMetadata;
+import com.facebook.presto.testing.TestingPageSinkProvider;
+import com.facebook.presto.testing.TestingSplitManager;
+import com.facebook.presto.testing.TestingTransactionHandle;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.facebook.presto.tpch.TpchPlugin;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestBeginQuery
+        extends AbstractTestQueryFramework
+{
+    private TestMetadata metadata;
+
+    protected TestBeginQuery()
+            throws Exception
+    {
+        super(createQueryRunner());
+    }
+
+    private static QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("test")
+                .setSchema("default")
+                .build();
+        return new DistributedQueryRunner(session, 1);
+    }
+
+    @BeforeClass
+    public void setUp()
+            throws Exception
+    {
+        metadata = new TestMetadata();
+        queryRunner.installPlugin(new TestPlugin(metadata));
+        queryRunner.installPlugin(new TpchPlugin());
+        queryRunner.createCatalog("test", "test", ImmutableMap.of());
+        queryRunner.createCatalog("tpch", "tpch", ImmutableMap.of());
+    }
+
+    @AfterClass(alwaysRun = true)
+    private void tearDown()
+    {
+        queryRunner.close();
+    }
+
+    @BeforeMethod
+    public void beforeMethod()
+            throws Exception
+    {
+        metadata.clear();
```

Comment:
I do `reset()` after each query. It resets the begin/end call counters. I will rename `reset()` to `resetCounters()` to make it more clear.